src/components/Searcher.js

```diff
@@ -495,6 +498,37 @@ class Searcher extends React.Component<SearcherDefaultProps, SearcherProps, Sear
     });
   }
 
+  /**
+   * Fetches the signal data from the first document of the response.
+   * If the document or the signal does not exist, and createNewIfNotExisting = true,
+   * a new SignalData object is created with default values populated.
+   */
+  getDefaultQuerySignal = (createNewIfNotExisting?: boolean = false) => {
+    const queryDocuments = this.state.response ? this.state.response.documents : null;
+    const querySignal = queryDocuments && queryDocuments.length >= 1 ? queryDocuments[0].signal : null;
+    if (querySignal) {
+      if (!querySignal.relevancyModelVersion) {
+        querySignal.relevancyModelVersion = 1;
+      }
+      return querySignal;
+    }
+    if (createNewIfNotExisting) {
+      const savedUser = AuthUtils.getSavedUser();
+      const defaultSignalData = new SignalData();
+      if (savedUser) {
+        defaultSignalData.locale = 'en';
+        defaultSignalData.principal = `${AuthUtils.config.ALL.defaultRealm}:${savedUser.fullName}:${savedUser.userId}`;
+        defaultSignalData.query = this.state.query;
+        defaultSignalData.queryTimestamp = this.state.queryTimestamp;
+        defaultSignalData.relevancyModelName = 'default';
```

Comment:
The relevancy model used for the query is also in the searcher's state... we should try to get it from there instead of using "default" since customers will expect this to be correct.

Reply:
Sounds good. However, when there are no search results returned, the `relevancyModels` in the state is not populated. So, I put a check to see if the value is present in the state. If not, I have defaulted it to "default".

Reply:
The property `relevancymodelnames` in the state should generally be the currently selected relevancy model(s) (well, usually just one) that are added to the query request when searching. If you look at the code in the `Searcher`'s `getQueryRequest()` method, you can refactor the bit that decides what to put for the "relevancyModelNames" REST parameter into a separate method and call it from both `getQueryRequest()` and `getDefaultQuerySignal()`.